include/modem/modem_info.h

```diff
@@ -298,6 +311,15 @@ int modem_info_name_get(enum modem_info info, char *name);
  */
 enum at_param_type modem_info_type_get(enum modem_info info);
```

Comment:
You could also add `__deprecated` to the function declaration itself.
```suggestion
__deprecated enum at_param_type modem_info_type_get(enum modem_info info);
```